packages/mirinae/src/data-display/collapsible/collapsible-panel/composables/use-text-clamper.ts

```diff
@@ -0,0 +1,223 @@
+import { reactive } from 'vue';
+
+interface TextClamper {
+    state: { isOverflow: boolean; }
+    runTextClamper: (isCollapsed: boolean, contentRef : HTMLElement | null, fakeTextRef? : HTMLElement | null) => void;
+}
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion, no-continue, no-restricted-syntax */
+export const useDeepTextClamper = (lineClamp : number):TextClamper => {
+    const state = reactive({
+        isOverflow: false,
+        clampedElement: null as null|HTMLElement,
+        originalStyles: null as null | Partial<CSSStyleDeclaration>,
+        isClampApplied: false,
+        hiddenNodesMap: new Map<Node, string>(),
+    });
+
+    const calculateLineCount = (element: HTMLElement) : number => {
+        const computedStyle = window.getComputedStyle(element);
+        const paddingTop = parseFloat(computedStyle.paddingTop) || 0;
+        const paddingBottom = parseFloat(computedStyle.paddingBottom) || 0;
+        const totalHeight = (element.scrollHeight || element.getBoundingClientRect().height) - paddingTop - paddingBottom;
+
+        if (totalHeight === 0) {
+            return 0;
+        }
+
+        let lineHeight = parseFloat(computedStyle.lineHeight);
+
+        // eslint-disable-next-line no-restricted-globals
+        if (isNaN(lineHeight)) {
+            const fontSize = parseFloat(computedStyle.fontSize);
+            lineHeight = fontSize * 1.2;
+        }
```

Comment:
nitpick: Consider using Number.isNaN instead of isNaN for better type safety and to avoid implicit type coercion.

Reply:
Thank you for the good point. I'll take it